lib/school/protocol.ex

```diff
@@ -0,0 +1,24 @@
+alias School.Students.Student
+alias School.Teachers.Teacher
+alias School.Repo
+
+defprotocol School.Person do
+  def show(person)
+end
+
+defimpl School.Person, for: Student do
+  def show(student) do
+    student.courses
+    |> Enum.filter(fn course -> course.semester == :first end)
+    |> length()
```

Comment:
show function should return the gender, address, date of birth and the number of courses, but you're only returning the the number of courses here